src/ForgeVTT.mjs

```diff
@@ -1327,25 +1328,48 @@ export class ForgeVTT {
       }
       return `<img src="${ForgeVTT.FORGE_URL}${img}" width="24" style="border: 0px; vertical-align:middle;"/>`;
     };
-    const buttons = options
-      .map((p) => `<div><button data-join-as="${p.id}">${p.name} ${roleToImg(p.role)}</button></div>`)
-      .join("");
+
     // Close the main menu if it was open
     ui.menu.close();
-    new Dialog(
-      {
-        title: "Join Game As",
-        content: `<p>Select a player to re-join the game as : </p>${buttons}`,
-        buttons: {},
-        render: (html) => {
-          for (const button of ForgeVTT.ensureIsJQuery(html).find("button[data-join-as]")) {
-            const as = button.dataset.joinAs;
-            $(button).on("click", () => (window.location.href = `/join?as=${as}`));
-          }
+
+    if (ForgeVTT.isFoundryNewerThan("11")) {
```

Comment:
suggestion: split into two methods

This is getting bloated, and splitting it into two methods (even if they call only one function inside) would make it easier for reading in this function - or make a "factory" method `getJoinAsApplication(...)` that would return an application you can then `.render()`